README.md

```diff
@@ -15,26 +15,75 @@ Once you have implemented the `sa` program (and tested it to the best of your ab
 
 # Report
 
-Jeg forventer at blive færdig med rapporten i dag (lørdag) <br>
--Gaia
 
+## Algorithm
 
+We used the construction algorithm based on raddix sort. 
+The algorithm has a running time of O(n²), which we will compare to the running time of our implementation below. 
 
-## Algorithm
+## Insights you may have had while implementing the algorithm
+I [Gaia] have learned a lot about C and how it compares to Python. 
+Since I started writing the algorithms in Python, the differences became clear, 
+when translated to C. 
+Some consepts I have a better understanding of include pointers, 
+structs, iterator pattern, header files, char as int, and strings. <br>
+Some insights regarding Windows vs. Linux became clear, when we experienced problem, as described below. 
 
-*Which algorithm did you use to construct the suffix array. What is its running time?*
+## Problems encountered if any
+Due to privious projects not being finished, this project had to be made alongside those. 
+We chose to focus on radix sort rather than more complicated algorithms like skew and SAIS to have more time.
+Radix sort and binary search were not problematic in themselves, which gave us time to focus on the implementation.
 
+The group added a new member, Gaia, who had almost no prior experience with writing C. 
+Because of this, the first meeting of Gaia and Jonathan was focused on getting Gaia ready 
+by running code from project 1. Running this on a Windows mashine (priviously only run on Linux and Mac) 
+uncovered two problems in the code. 
+1. Since Windows saves line breaks as \r\n a line break in files read with "r" will be counted as two chars,
+which will make the string length longer than expected. This was solved by reading the file in binary mode "rb". 
+2. When making the strict border array, the code was missing a check for whether bax[i]==0. Without this, the code tried 
+to find index -1 of bax. By coincidence index -1 pointed to a place in memory which contained 0, the desired value, 
+in every test. Probably since there was some relation between the location of bax and some other value. 
+In Linux that is. When run on Windows, index -1 gave a very small negative number. This is not a problem with Windows, 
+but an implementation error only discovered, because we ran it on Windows. 
```

Comment:
Indexing outside of an allocated block is undefined behaviour (which is rarely good). The same for even having a pointer *point* there, even if you don't dereference it. Be careful with that.